internal/cmp/router.go

```diff
@@ -0,0 +1,137 @@
+// (C) Copyright 2021 Hewlett Packard Enterprise Development LP
+
+package cmp
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/pkg/client"
+	"github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/pkg/models"
+	"github.com/HewlettPackard/hpegl-vmaas-terraform-resources/internal/utils"
+	"github.com/tshihad/tftags"
+)
+
+type router struct {
+	routerClient *client.RouterAPIService
+}
+
+func newRouter(routerClient *client.RouterAPIService) *router {
+	return &router{
+		routerClient: routerClient,
+	}
+}
+
+func (r *router) Read(ctx context.Context, d *utils.Data, meta interface{}) error {
+	routerID := d.GetID()
+	routerResp, err := utils.Retry(ctx, meta, func(ctx context.Context) (interface{}, error) {
+		return r.routerClient.GetSpecificRouter(ctx, routerID)
+	})
+	if err != nil {
+		return err
+	}
+	d.SetID(routerResp.(models.GetSpecificRouterResp).NetworkRouter.ID)
+
+	return nil
+}
+
+func (r *router) Create(ctx context.Context, d *utils.Data, meta interface{}) error {
+	createReq := models.CreateRouterRequest{}
+	if err := tftags.Get(d, &createReq.NetworkRouter); err != nil {
+		return err
+	}
+	// align createReq and fill json related fields
+	r.routerAlignRouterRequest(ctx, meta, &createReq)
+
+	resp, err := utils.Retry(ctx, meta, func(ctx context.Context) (interface{}, error) {
+		return r.routerClient.CreateRouter(ctx, createReq)
+	})
+	if err != nil {
+		return err
+	}
+	routerResp := resp.(models.CreateRouterResp)
+	if !routerResp.Success {
+		return fmt.Errorf("got success = 'false' while creating router")
+	}
+	d.SetID(routerResp.ID)
+
+	return nil
+}
+
+func (r *router) Update(ctx context.Context, d *utils.Data, meta interface{}) error {
```

Comment:
Router update is not supported? or is it out of scope of the story?

Reply:
It is another story